sphinx/builders/linkcheck.py

```diff
@@ -99,61 +110,59 @@ def process_result(self, result: CheckResult) -> None:
                     'info': result.message}
         self.write_linkstat(linkstat)
 
-        if result.status == 'unchecked':
-            return
-        if result.status == 'working' and result.message == 'old':
-            return
+        match result.status:
```

Comment:
We can't adopt `match` blocks as we still support Python 3.9 until at least April.

Reply:
Do you think it's worth merging the `StrEnum` usage without the `match` statements?  I'm on the fence about it, so I'd be OK waiting until Py3.9 is our baseline.

Reply:
> Do you think it's worth merging the `StrEnum` usage without the `match` statements?

`Enum` with `match` is always worth it. You can get the same result by using an `Enum` that inherits from `str`, see [this example by Ethan Furman][1] and the [canonical SO Q][2].

[1]: https://stackoverflow.com/a/69457785
[2]: https://stackoverflow.com/a/58608362